dal/db_race_test.go

```diff
@@ -1,95 +1,97 @@
 package dal
 
 import (
+	"sync"
 	"testing"
 
 	"gotest.tools/v3/assert"
 )
 
+const (
+	batch               = 1000
+	workers             = 5
+	testTimestamp int64 = 123
+)
+
 func Test_DbRace(t *testing.T) {
 	db, err := NewDB(":memory:")
 	assert.NilError(t, err)
 	err = db.InitDB()
 	assert.NilError(t, err)
 
-	const batch = 1000
-	const workers = 5
-	done := make(chan bool, workers)
-
+	// Test 1: Parallel inserts
+	var wg1 sync.WaitGroup
 	for i := uint64(0); i < workers; i++ {
-		// TODO: use WorkGroup
-		go insertManySignReq(t, done, db, i*batch, (i+1)*batch)
-	}
-	for i := 0; i < workers; i++ {
-		<-done
+		wg1.Add(1)
+		go func(i uint64) {
+			defer wg1.Done()
+			insertManySignReq(t, db, i*batch, (i+1)*batch)
+		}(i)
 	}
+	wg1.Wait()
 
 	var srID uint64 = 1
 	sr, err := db.GetIkaSignRequestByID(srID)
 	assert.NilError(t, err)
 	assert.Assert(t, sr != nil)
-	assert.Equal(t, int64(123), sr.Timestamp)
+	assert.Equal(t, testTimestamp, sr.Timestamp)
 
-	row := db.conn.QueryRow(`
-SELECT COUNT(*) FROM ika_sign_requests`)
+	row := db.conn.QueryRow(`SELECT COUNT(*) FROM ika_sign_requests`)
 	var count int
 	err = row.Scan(&count)
 	assert.NilError(t, err)
 	assert.Equal(t, batch*workers, count)
 
-	// test 2
-	// try parallel update
-
+	// Test 2: Parallel updates
+	var wg2 sync.WaitGroup
 	for i := uint64(0); i < workers; i++ {
-		go loopIncrementIkaSRTimestamp(t, done, db, batch, srID)
-	}
-	for i := 0; i < workers; i++ {
-		<-done
+		wg2.Add(1)
```

Comment:
Did you find out why we can't pass `wg` to the function, even as a pointer?

Reply:
we can pass it as a pointer. However then we can only call `wg.Done()` inside the function itself. When we try to call also the `wg.Add(1)` inside the function it creates issues. Its probably due to how the function code is executed and that there is a slight delay. So we might reach the `wg.Wait()` before  all the code actually finished

Reply:
Thats why i decided to keep the wg outside of the function. I think its better, rather than doing wg.Add outside and wg.Done inside

Reply:
actually, `wg.Add` should be outside (that was the initial error), and we can do wg.Done inside